src/Mod/Draft/DraftGui.py

```diff
@@ -1893,8 +1893,7 @@ def getXPM(self,iconname,size=16):
         return str(a)
 
     def togglesnap(self):
-        if hasattr(FreeCADGui,"Snapper"):
-            FreeCADGui.Snapper.toggle()
+        FreeCADGui.doCommand('FreeCADGui.runCommand("Draft_Snap_Lock")')
```

Comment:
Is this the best way of doing it? I am not entirely sure what class calls which. To run this command, it must be defined first, but `DraftGui` is one of the early modules loaded. So, it's like calling something that will be defined later.

If this is the case, then `DraftGui` must be loaded after the `gui_snaps` commands are loaded.

As I can see, `Draft_Snap_Lock` basically calls `Gui.Snapper.masterbutton.toggle()`, so it looks very similar to `Gui.Snapper.toggle()`.

Anyway, it's just a comment. As I said, `DraftGui` is quite complex because it is both the DraftToolBar, the snap toolbar, and the task panel interface for most commands.

Reply:
> As I can see, Draft_Snap_Lock basically calls Gui.Snapper.masterbutton.toggle(), so it looks very similar to Gui.Snapper.toggle().
This was the previous behaviour. With proposed changes `Draft_Snap_Lock ` just calls `Gui.Snapper.toggle_snap('Lock')` and sync the interface (toolbar and statusbar).
I wish the default behaviour to always interact with Snapper through Snaps commands to set the active_snaps.
Do you think we should modify something in the order the modules are loaded?

Reply:
I'm just not sure how it works really. I've never investigated deeply this behavior. But ideally we should try split DraftGui in discrete parts as much as possible.